src/ember-app.js

```diff
@@ -344,39 +350,40 @@ class EmberApp {
     }
 
     let manifest;
+    let schemaVersion;
     let pkg;
 
     try {
       pkg = JSON.parse(file);
       manifest = pkg.fastboot.manifest;
+      schemaVersion = pkg.fastboot.schemaVersion;
     } catch (e) {
       throw new Error(`${pkgPath} was malformed or did not contain a manifest. Ensure that you have a compatible version of ember-cli-fastboot.`);
     }
 
-    var appFiles = [];
-    if (manifest.appFiles) {
-      debug("reading array of app file paths from manifest");
-      manifest.appFiles.forEach(function(appFile) {
-        appFiles.push(path.join(distPath, appFile));
-      });
-    } else if (manifest.appFile) {
-      // TODO : remove after Fastboot 1.0
-      debug("reading app file path from manifest");
-      appFiles = [path.join(distPath, manifest.appFile)];
+    if (schemaVersion !== LATEST_SCHEMA_VERSION && schemaVersion !== NEXT_SCHEMA_VERSION) {
```

Comment:
I think we should only fail if the version of the schema in the manifest is *newer* than the current schema version. For example, if the manifest version is 57 and the latest the FastBoot server knows about is 13, then there may be information in the manifest we don't understand. In that case, we should fail. But if the version in the manifest is *older*, we know its characteristics and should be able to transform it into the latest manifest schema. That way people can still run older builds while keeping their FastBoot servers up-to-date with the latest version.